cmd/operator-sdk/run/local.go

```diff
@@ -76,65 +85,40 @@ func (c runLocalArgs) run() error {
 	return projutil.ErrUnknownOperatorType{}
 }
 
+// runGo will run the project locally for the Go Type projects
 func (c runLocalArgs) runGo() error {
-	projutil.MustInProjectRoot()
-	absProjectPath := projutil.MustGetwd()
-	projectName := filepath.Base(absProjectPath)
-	outputBinName := filepath.Join(scaffold.BuildBinDir, projectName+"-local")
-	if runtime.GOOS == "windows" {
-		outputBinName += ".exe"
-	}
-	if err := c.buildLocal(outputBinName); err != nil {
-		return fmt.Errorf("failed to build operator to run locally: %v", err)
-	}
-
-	args := []string{}
-	if c.operatorFlags != "" {
-		extraArgs := strings.Split(c.operatorFlags, " ")
-		args = append(args, extraArgs...)
+	// Build the project and generate binary that will be executed
+	binName, err := c.generateBinary()
+	if err != nil {
+		return err
 	}
-
-	var dc *exec.Cmd
-
+	// Get the args that will be used to exec the binary.
+	// Users are allowed to use the flag operator-flags to pass any value that they may wish
+	args := c.buildArgsBasedOnOperatorFlag()
```

Comment:
Nit: Maybe shorten the name a bit.
```suggestion
	args := c.argsFromOperatorFlags()
```